src/iexamine.cpp

```diff
@@ -3497,6 +3498,80 @@ void iexamine::climb_down( player &p, const tripoint &examp )
     g->m.creature_on_trap( p );
 }
 
+void iexamine::autodoc( player &p, const tripoint &examp ) {
+    enum options {
+        CONFIGURE,
+        INSTALL_CBM,
+        CANCEL,
+    };
+
+    std::string status = uistate.iexamine_autodoc_config ? _( "Online" ) :
+                         _( "Configuration required" );
+
+    uimenu amenu;
+    amenu.selected = 0;
+    amenu.text = string_format( _( "Autodoc Mk. XI.  Status: %s.  Please choose operation." ), status );
+    amenu.addentry( CONFIGURE, true, 'c', str_to_illiterate_str( _( "Configure Autodoc." ) ) );
+    amenu.addentry( INSTALL_CBM, uistate.iexamine_autodoc_config, 'i',
+                    str_to_illiterate_str( _( "Choose Compact Bionic Module to install." ) ) );
+
+    amenu.addentry( CANCEL, true, 'q', str_to_illiterate_str( _( "Do nothing." ) ) );
+
+    amenu.query();
+
+    bool adjacent_couch = false;
+    bool in_position = false;
+    for( const auto &couch_loc : g->m.points_in_radius( examp, 1, 0 ) ) {
+        if( g->m.furn( couch_loc ).obj().id == "f_autodoc_couch" ) {
+            adjacent_couch = true;
+            if( p.pos() == couch_loc ) {
+                in_position = true;
+            }
+        }
+    }
+
+    switch( static_cast<options>( amenu.ret ) ) {
+        case CONFIGURE:
+            uistate.iexamine_autodoc_config = true;
```

Comment:
This is a global variable, setting it will affect *all* couches all over the world. It's also not saved anywhere, so it's reset whenever the player loads a game.

Reply:
> This is a global variable, setting it will affect all couches all over the world.

Can you recommend something to achieve the desired goal, but only for the coach player interacts with?

Reply:
Prob you need two furnitures: f_autodoc_unconfigured, f_autodoc_configured.

Reply:
I'd rather not add another furniture. Anybody know, if there is a way to add flags to furniture during runtime?